integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/mock/DistroXClusterFaultTolerantUpscaleTest.java

```diff
@@ -0,0 +1,223 @@
+import com.sequenceiq.it.cloudbreak.client.SdxTestClient;
+import com.sequenceiq.it.cloudbreak.context.Description;
+import com.sequenceiq.it.cloudbreak.context.MockedTestContext;
+import com.sequenceiq.it.cloudbreak.context.RunningParameter;
+import com.sequenceiq.it.cloudbreak.dto.distrox.DistroXTestDto;
+import com.sequenceiq.it.cloudbreak.dto.distrox.cluster.DistroXClusterTestDto;
+import com.sequenceiq.it.cloudbreak.dto.distrox.cluster.clouderamanager.DistroXClouderaManagerTestDto;
+import com.sequenceiq.it.cloudbreak.dto.distrox.image.DistroXImageTestDto;
+import com.sequenceiq.it.cloudbreak.dto.distrox.instancegroup.DistroXNetworkTestDto;
+import com.sequenceiq.it.cloudbreak.exception.TestFailException;
+import com.sequenceiq.it.cloudbreak.testcase.mock.clouderamanager.AbstractClouderaManagerTest;
+
+public class DistroXClusterFaultTolerantUpscaleTest extends AbstractClouderaManagerTest {
+
+    private static final int SUCCESSFUL_UPSCALE_NODE_COUNT = 10;
+
+    private static final int FAILING_NODE_COUNT = 2;
+
+    private static final String UPSCALED_INSTANCE_GROUP_NAME = "worker";
+
+    private static final String IMAGE_CATALOG_ID = "f6e778fc-7f17-4535-9021-515351df3691";
+
+    private static final String CM_FOR_DISTRO_X = "cm4dstrx";
+
+    private static final String CLUSTER_KEY = "cmdistrox";
+
+    private static final String DIX_IMG_KEY = "dixImg";
+
+    private static final String DIX_NET_KEY = "dixNet";
+
+    private static final Duration POLLING_INTERVAL = Duration.of(3000, ChronoUnit.MILLIS);
+
+    @Inject
+    private BlueprintTestClient blueprintTestClient;
+
+    @Inject
+    private DistroXTestClient distroXClient;
+
+    @Inject
+    private SdxTestClient sdxTestClient;
+
+    @Test(dataProvider = TEST_CONTEXT_WITH_MOCK)
+    @Description(
+            given = "there is a running DistroX cluster",
+            when = "an upscale is called, but some of the new nodes are unreachable",
+            then = "the cluster should be AVAILABLE in a partially upscaled state")
+    public void testUpscaleWithFewNodesBeingUnreachable(MockedTestContext testContext, ITestContext testNgContext) {
+        DistroXStartStopTestParameters params = new DistroXStartStopTestParameters(testNgContext.getCurrentXmlTest().getAllParameters());
+        String stack = resourcePropertyProvider().getName();
+        createDatalake(testContext);
+
+        MinionIpAddressesResponse minionIpAddressesResponse = getMockResponseForIpAddrsSaltCall();
+        Set<String> saltBodyFilters = getSaltBodyFilters();
+
+        DistroXTestDto currentContext = setUpContextForUpscale(testContext, stack);
+        currentContext
+                .mockSalt().run().post().bodyContains(saltBodyFilters, 1)
+                .thenReturn(minionIpAddressesResponse, null, 200, 0, null)
+                .when(distroXClient.scale(params.getHostgroup(), SUCCESSFUL_UPSCALE_NODE_COUNT))
+                .awaitForFlow(RunningParameter.waitForFlowFail())
```

Comment:
I think these two steps can be merged to 
```
.awaitForFlow(RunningParameter.waitForFlowFail(), key(stack).withPollingInterval(POLLING_INTERVAL))
```
after waiting for a successful scale, I think we do not need to wait in another round for `STACK_AVAILABLE` state as well.

This is true in case of every occurence of this.

Reply:
If we don't wait for STACK_AVAILABLE state, the verification ends up running too soon, when none of the new nodes from the upscale are present, resulting in an error during verification.
The cluster should be in an AVAILABLE state soon after the flow failed, but just not soon enough for the new nodes to be present during verification.